e2e/app/page/base-page.ts

```diff
@@ -39,187 +38,16 @@ export default abstract class BasePage {
-      const isVisible = elem.offsetWidth || elem.offsetHeight || elem.getClientRects().length;
-      return !isVisible;
-    }, {}, cssSelector);
-  }
-
-  /**
-   * Wait for the elements count to be a particular value
-   * @param {string} cssSelector - The css selector for the element
-   * @param {number} expectedCount - The expected number of elements
-   */
-  async waitForNumberElements(cssSelector: string, expectedCount: number) {
-    return await this.page.waitForFunction((selector, count) => {
-      return document.querySelectorAll(selector).length === count;
-    }, {}, cssSelector, expectedCount);
-  }
-
-  /**
-   * Wait for the elements count to be greater than a particular value
-   * @param {string} cssSelector - The css selector for the element
-   * @param {number} greaterThanCount - The expected number of elements
-   */
-  async waitForNumberElementsIsBiggerThan(cssSelector: string, greaterThanCount: number) {
-    return await this.page.waitForFunction((selector, count) => {
-      return document.querySelectorAll(selector).length > count;
-    }, {}, cssSelector, greaterThanCount);
-  }
-
-  /**
-   * Look for visible texts on the page.
-   * @param {string} texts Partial texts to look for
-   */
-  async waitForTextExists(texts: string) {
-    return await this.page.waitForFunction(matchText => {
-      const bodyText = document.querySelector('body').innerText;
-      return bodyText.includes(matchText)
-    }, {}, texts);
-  }
-
-  /**
-   * Wait for the element found from the selector has a particular attribute value pair
-   * @param {string} cssSelector - The selector for the element on the page
-   * @param {string} attribute - The attribute name
-   * @param {string} value - The attribute value to match
-   */
-  async waitUntilContainsAttributeValue(cssSelector, attribute, value) {
-    return await this.page.waitForFunction((selector, attributeName, attributeValue) => {
-      const element = document.querySelector(selector);
-      return element.attributes[attributeName] && element.attributes[attributeName].value === attributeValue;
-    }, {}, cssSelector, attribute, value);
-  }
-
-  /**
-   * Wait for exact text to match.
-   * @param page
-   * @param cssSelector
-   * @param expectedText
-   */
-  async waitForText(cssSelector: string, expectedText: string) {
-    await this.page.waitForFunction( (css, expText) => {
-      const t = document.querySelector(css);
-      if (t !== undefined) {
-        return t.innerText === expText;
-      }
-    }, {}, cssSelector, expectedText);
-
-    await this.page.waitForSelector(cssSelector, { visible: true });
+    return this._page.$eval(`${cssSelector}`, elem => elem.textContent.trim())
   }
 
   /**
    * Find texts in DOM.
    * @param txt
    */
   async findText(txt: string): Promise<boolean> {
-    const indx = await (await this.page.content()).indexOf(txt);
+    const indx = await (await this._page.content()).indexOf(txt);
```

Comment:
Please rename this to `containsText()` so it looks more like a predicate. You could also make a small `FindResults` class that has a boolean for found and an index for the position (if found). Or just stick with the integer return value and call it `findStartIndex(text: string): Promise<integer?>` (if that's allowed).